src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyStringComparison.cs

```diff
@@ -82,6 +84,18 @@ public override void Initialize(AnalysisContext analysisContext)
                         return;
                     }
 
+                    if (iqueryableType != null)
+                    {
+                        var ancestorInvocation = invocationExpression.GetAncestor<IInvocationOperation>(OperationKind.Invocation);
```

Comment:
We could consider moving this block check at the time where we report if there is any performance concern with getting up the operation tree.

Reply:
This seems reasonable, but I am wondering if we can do a more targeted fix. For example, I am not sure if this will be behave correctly when crossing lambda/local function boundary.